cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/NamespaceExistenceVerifier.java

```diff
@@ -41,9 +41,7 @@
   public void ensureExists(NamespaceId namespaceId) throws NotFoundException {
     if (!NamespaceId.SYSTEM.equals(namespaceId)) {
       try {
-        if (namespaceQueryAdmin.get(namespaceId.toId()) == null) {
-          throw new NamespaceNotFoundException(namespaceId.toId());
-        }
+        namespaceQueryAdmin.exists(namespaceId.toId());
```

Comment:
ok. the contract of this method though is that it should throw a NamespaceNotFoundException if the namespace does not exist. I don't think that is obeyed here.